File: 백준/Silver/10828. 스택/스택.py
import sys

def push(stack, x):
    stack.append(x)

def pop(stack):
    if not stack:
        return -1
    return stack.pop()

def size(stack):
    return len(stack)

def empty(stack):
    return 1 if not stack else 0

def top(stack):
    if not stack:
        return -1
    return stack[-1]

if __name__ == "__main__":
    N = int(sys.stdin.readline())
    stack = []

    for _ in range(N):
        command = sys.stdin.readline().split()
        if command[0] == "push":
            push(stack, int(command[1]))
        elif command[0] == "pop":
            print(pop(stack))
        elif command[0] == "size":
            print(size(stack))
        elif command[0] == "empty":
            print(empty(stack))
        elif command[0] == "top":
            print(top(stack))

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();  // 스택선언
        while (N > 0) {

            String[] s = br.readLine().split(" ");
            String a;
            int b = 0;
            if (s.length > 1) {
                a = s[0];
                b = Integer.parseInt(s[1]);
            } else {
                a = s[0];
            }

            if(a.equals("push") ) {
                stack.push(b);
            }
            else if (a.equals("top")) {
                if(stack.isEmpty()) {
                    System.out.println(-1);
                }
                else {
                    System.out.println(stack.peek());
                }
            }
            else if(a.equals("pop")) {
                if (stack.isEmpty()){
                    System.out.println(-1);
                }
                else {
                    System.out.println(stack.pop());
                }
            }
            else if(a.equals("size")) {
                System.out.println(stack.size());
            }
            else {
                if (stack.isEmpty()) {
                    System.out.println(1);
                }
                else {
                    System.out.println(0);
                }

            }
            N--;
        }
    }
}
